pkg/feature/manifest.go

```diff
@@ -7,31 +7,177 @@ import (
+}
+
+// Ensure baseManifest implements the Manifest interface.
+var _ Manifest = (*baseManifest)(nil)
+
+func (b *baseManifest) Process(_ interface{}) ([]*unstructured.Unstructured, error) {
+	manifestFile, err := b.fsys.Open(b.path)
+	if err != nil {
+		return nil, err
+	}
+	defer manifestFile.Close()
+
+	content, err := io.ReadAll(manifestFile)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read file: %w", err)
+	}
+	resources := string(content)
+
+	var objs []*unstructured.Unstructured
+
+	objs, err = convertToUnstructureds(resources, objs)
+	if err != nil {
+		return nil, err
+	}
+	return objs, nil
+}
+
+type templateManifest struct {
 	name,
-	path,
-	processedContent string
-	template,
+	path string
 	patch bool
-	fsys fs.FS
+	fsys  fs.FS
+}
+
+func (t *templateManifest) isPatch() bool {
+	return t.patch
+}
+
+func (t *templateManifest) Process(data interface{}) ([]*unstructured.Unstructured, error) {
+	manifestFile, err := t.fsys.Open(t.path)
+	if err != nil {
+		return nil, err
+	}
+	defer manifestFile.Close()
+
+	content, err := io.ReadAll(manifestFile)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create file: %w", err)
+	}
+
+	tmpl, err := template.New(t.name).Funcs(template.FuncMap{"ReplaceChar": ReplaceChar}).Parse(string(content))
+	if err != nil {
+		return nil, fmt.Errorf("failed to create file: %w", err)
+	}
+
+	var buffer bytes.Buffer
+	if err := tmpl.Execute(&buffer, data); err != nil {
+		return nil, err
+	}
+
+	resources := buffer.String()
+	var objs []*unstructured.Unstructured
+
+	objs, err = convertToUnstructureds(resources, objs)
+	if err != nil {
+		return nil, err
+	}
+	return objs, nil
+}
+
+// Ensure templateManifest implements the Manifest interface.
+var _ Manifest = (*templateManifest)(nil)
+
+// kustomizeManifest supports paths to kustomization files / directories containing a kustomization file
+// note that it only supports to paths within the mounted files ie: opt/manifests.
+type kustomizeManifest struct {
+	name,
+	path string // path is to the directory containing a kustomization.yaml file within it or path to kust file itself
+	fsys filesys.FileSystem
+}
+
+func (k *kustomizeManifest) isPatch() bool {
+	return false
 }
 
-func loadManifestsFrom(fsys fs.FS, path string) ([]manifest, error) {
-	var manifests []manifest
+func (k *kustomizeManifest) Process(data interface{}) ([]*unstructured.Unstructured, error) {
+	kustomizer := krusty.MakeKustomizer(krusty.MakeDefaultOptions())
+	var resMap resmap.ResMap
+	resMap, resErr := kustomizer.Run(k.fsys, k.path)
+
+	if resErr != nil {
+		return nil, fmt.Errorf("error during resmap resources: %w", resErr)
+	}
+
+	targetNs := getTargetNs(data)
+	if targetNs == "" {
+		return nil, fmt.Errorf("error grabbing targetNs from feature spec")
+	}
+
+	if err := plugins.ApplyNamespacePlugin(targetNs, resMap); err != nil {
+		return nil, err
+	}
+
+	// Todo: for now, this only supports applying labels to manifests source.Name when source.type==Component
```

Comment:
What is left in the actual `TODO` scope here? :) What we need to do more?

Reply:
this note is moreso for future work - considering how DSCI.monitoring uses kustomize to create resources for monitoring (with separate component names passed into `deploy.Deploy...()`) , thinking about how we handle these cases

Reply:
Can we convert it to a sub-task of https://issues.redhat.com/browse/RHOAIENG-3261?

Reply:
done!